### DISCUSSION TITLE
Do not fail on nonexistent metadata files

### DISCUSSION THREAD

**Discussion by avidseeker** [jgm/pandoc#9813] [Ideas]
I have `defaults.yaml` that I use for documents in specific project structure:

```yml
metadata-files:
  - ../meta.yml
  - ../../meta.yml
```

and I would like to use the same defaults file for files outside that project. Can pandoc just assume that metadata was not provided instead of failing on `Could not find metadata file ../metadata.yml`?

**Comment by tarleb** [comment]
A better method might be to use a filter that looks for metadata files, reads them with `pandoc.read`, and then adds the result to the document.

``` lua
local function find_metadata_files()
  -- implement
end

function Pandoc (doc)
  for _, filename in ipairs(find_metadata_files()) do
    doc = doc .. pandoc.read(io.open(filename):read 'a', 'markdown')
  end
  return doc
end
```

**Reply by avidseeker** [reply]
Metadata are yaml files, doesn't that adds them as markdown files? If that is changed to yamls then `Error: Unknown input format yaml`.

**Comment by jgm** [comment]
I think that failing is the right thing to do in this case.  The filter may be a good workaround for you.
